### PR TITLE
[ASP.NET Core] Remove OTEL_SEMCONV_STABILITY_OPT_IN

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Fixes #
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+* Removed support for `OTEL_SEMCONV_STABILITY_OPT_IN` environment variable. The
+  library will now emit
+  [stable](https://github.com/open-telemetry/semantic-conventions/tree/v1.23.0/docs/http)
+  semantic conventions by default.
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:8]: Rephrase this and remove "by default" to something like "The library will now emit only the stable semantic conventions".

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md]: done

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/IncomingRequestsCollectionsIsAccordingToTheSpecTests.cs [hunk 2]
```diff
@@ -39,114 +39,106 @@ public IncomingRequestsCollectionsIsAccordingToTheSpecTests(WebApplicationFactor
     }
 
     [Theory]
-    [InlineData("/api/values", null, "user-agent", 503, "503")]
-    [InlineData("/api/values", "?query=1", null, 503, null)]
+    [InlineData("/api/values", null, "user-agent", 200, null)]
```
Comment (utpilla) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/IncomingRequestsCollectionsIsAccordingToTheSpecTests.cs:42]: Why update the test cases? Do you simply want to add test cases for `200`?

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs [hunk 3]
```diff
@@ -497,39 +360,7 @@ private static List<MetricPoint> GetMetricPoints(Metric metric)
...
-            }
-            else
-            {
-                Assert.Fail($"A metric for route '{expectedRoute}' was not found");
-            }
-        }
-    }
-
-    private static KeyValuePair<string, object>[] AssertMetricPoint_New(
+    private static KeyValuePair<string, object>[] AssertMetricPoint(
```
Comment (utpilla) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs:363]: Not related to this PR. Does this method need to return the attributes? It doesn't get used by the caller.

**Review by utpilla — APPROVED** [review]

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/IncomingRequestsCollectionsIsAccordingToTheSpecTests.cs (hunk 2 above)
Comment (vishweshbankwar) [reply test/OpenTelemetry.Instrumentation.AspNetCore.Tests/IncomingRequestsCollectionsIsAccordingToTheSpecTests.cs:42]: I did not update it in this PR. It was already there, it showing up due to file rename

Comment (vishweshbankwar) [reply test/OpenTelemetry.Instrumentation.AspNetCore.Tests/IncomingRequestsCollectionsIsAccordingToTheSpecTests.cs:42]: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4986

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs (hunk 3 above)
Comment (vishweshbankwar) [reply test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs]: good catch, fixed.

**Comment by TimothyMothra** [comment]
nit: I found a comment that's still referencing the older spec. Can this be updated to reference the stable 1.23.0?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/eec09c9fa54e746410deeb2b77592bc2bb50f27f/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs#L110

**Pull request merged**